### PR TITLE
OKTA-331690 .NET > Management Sdk > Add overload to add HOTP factor

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**
## Issue \#

https://oktainc.atlassian.net/browse/OKTA-331690

## Code

- [x] Unit test(s)
- [x] Implementation

## Additional Context

Added overloads and helper classes for HOTP Factor

### CODE REVIEW

File: src/Okta.Sdk/AddCustomHotpFactorOptions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Okta.Sdk
+{
+    /// <summary>
+    /// Custom HMAC-based One-Time Password (HOTP) factor options
+    /// </summary>
+    public class AddCustomHotpFactorOptions
```
Comment (bryanapellanes-okta) [inline src/Okta.Sdk/AddCustomHotpFactorOptions.cs:15]: Is this the established naming convention?  Specifically wondering about the verb prefix **Add**CustomHotpFactorOptions vs just CustomHotpFactorOptions.

File: src/Okta.Sdk/IUserFactorsClient.cs [hunk 2]
```diff
@@ -81,7 +81,16 @@ public partial interface IUserFactorsClient
...
+        Task<IUserFactor> AddFactorAsync(string userId, AddTotpFactorOptions totpFactorOptions, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Enrolls a user with a token:hotp factor
+        /// </summary>
+        /// <param name="userId">The user id</param>
+        /// <param name="hotpFactorOptions">The token:hotp factor options</param>
+        /// <param name="cancellationToken">The cancellation token</param>
+        /// <returns>The <see cref="IUserFactor"/> resource</returns>
+        Task<IUserFactor> AddFactorAsync(string userId, AddCustomHotpFactorOptions hotpFactorOptions, CancellationToken cancellationToken = default);
```
Comment (bryanapellanes-okta) [inline src/Okta.Sdk/IUserFactorsClient.cs:93]: I think I found the answer to my question 😞

**Review by bryanapellanes-okta — APPROVED** [review]
👍

**Review by laura-rodriguez — APPROVED** [review]
LGTM 👍 . 

I have created another JIRA to regenerate the rest of the code using the latest open API spec version. Once we have those changes too we can release a new version.

**Pull request merged**